File: src/main/java/io/github/dqualizer/dqlang/types/domain_architecture_mapping/Endpoint.java
package io.github.dqualizer.dqlang.types.domain_architecture_mapping;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.dqualizer.dqlang.archive.loadtesttranslator.dqlang.domainarchitecturemapping.Response;
import lombok.*;

import java.util.LinkedHashSet;
import java.util.Map;
import java.util.Set;

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class Endpoint {

    private String field;
    private String operation;
    @JsonProperty("path_variables")
    private Map<String, String> pathVariables;
    @JsonProperty("url_parameter")
    private Map<String, String> urlParameter;
    @JsonProperty("request_parameter")
    private Map<String, String> requestParameter;
    private Map<String, String> payload;
    private Set<Response> responses;
}

File: src/main/kotlin/dqualizer/dqlang/archive/loadtesttranslator/dqlang/domainarchitecturemapping/Endpoint.kt
package dqualizer.dqlang.archive.loadtesttranslator.dqlang.domainarchitecturemapping

import com.fasterxml.jackson.annotation.JsonProperty
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.ToString

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
class Endpoint {
    private val field: String? = null
    private val operation: String? = null

    @JsonProperty("path_variables")
    private val pathVariables: Map<String, String>? = null

    @JsonProperty("url_parameter")
    private val urlParameter: Map<String, String>? = null

    @JsonProperty("request_parameter")
    private val requestParameter: Map<String, String>? = null
    private val payload: Map<String, String>? = null
    private val responses: LinkedHashSet<Response>? = null
}
